routes/shopItem.js

```diff
@@ -36,16 +38,18 @@ shopItemRouter.put("/", (req, res, next) => {
             res.status(400).send();
             return;
         }
-        const isIdInOurDatabase = req.database.shopItems.findIndex((item) => {
-            return item.id === req.body.id;
+        db.run("UPDATE grocery_list SET name=$name, is_completed=$is_completed WHERE id=$id", {
+            $name: req.body.name,
+            $is_completed: req.body.isCompleted,
+            $id: req.body.id
+        }, function (err) {
+            if (err) {
+                console.log(err);
+                res.status(500).send();
+            } else {
+                res.status(200).send();
```

Comment:
If you still would like to keep the 404 behaviour, you can check this.changes here. If it is equals to 0, code 200. Otherwise 404